internal/pkg/cli/svc_override.go

```diff
@@ -120,6 +154,109 @@ func (o *overrideSvcOpts) validateCDKLang() error {
 		strings.Join(applyAll(validCDKLangs, strconv.Quote), ", "))
 }
 
+func (o *overrideSvcOpts) validateOrAskServiceName() error {
+	if o.name == "" {
+		return o.askServiceName()
+	}
+	return o.validateServiceName()
+}
+
+func (o *overrideSvcOpts) validateServiceName() error {
+	names, err := o.ws.ListServices()
+	if err != nil {
+		return fmt.Errorf("list services in the workspace: %v", err)
+	}
+	if !contains(o.name, names) {
+		return fmt.Errorf("service %q does not exist in the workspace", o.name)
+	}
+	return nil
+}
+
+func (o *overrideSvcOpts) askServiceName() error {
+	name, err := o.wsPrompt.Service("Which service's resources would you like to override?", "")
+	if err != nil {
+		return fmt.Errorf("select service name from workspace: %v", err)
+	}
+	o.name = name
+	return nil
+}
+
+func (o *overrideSvcOpts) validateOrAskIaCTool() error {
+	if o.iacTool == "" {
+		return o.askIaCTool()
+	}
+	return o.validateIaCTool()
+}
+
+func (o *overrideSvcOpts) validateIaCTool() error {
+	for _, valid := range validIaCTools {
+		if o.iacTool == valid {
+			return nil
+		}
+	}
+	return fmt.Errorf("%q is not a valid IaC tool: must be one of: %s",
+		o.iacTool,
+		strings.Join(applyAll(validIaCTools, strconv.Quote), ", "))
+}
+
+func (o *overrideSvcOpts) askResourcesToOverride() error {
+	if len(o.resources) != 0 {
+		return nil
+	}
+
+	buf := &closableStringBuilder{
+		Builder: new(strings.Builder),
+	}
+	pkgCmd, err := o.packageCmd(buf)
+	if err != nil {
+		return err
+	}
+	if err := pkgCmd.Execute(); err != nil {
+		return fmt.Errorf("generate CloudFormation template for service %q: %v", o.name, err)
+	}
+	msg := fmt.Sprintf("Which resources in %q would you like to override?", o.name)
+	resources, err := o.cfnPrompt.Resources(msg, "Resources:", "", buf.String())
+	if err != nil {
+		return err
+	}
+	o.resources = resources
+	return nil
+}
+
+func (o *overrideSvcOpts) newSvcPackageCmd(tplBuf stringWriteCloser) (executor, error) {
+	envs, err := o.cfgStore.ListEnvironments(o.appName)
+	if err != nil {
+		return nil, fmt.Errorf("list environments in application %q: %v", o.appName, err)
+	}
+	if len(envs) == 0 {
+		return nil, fmt.Errorf("no environments found in application %q", o.appName)
+	}
+	cmd, err := newPackageSvcOpts(packageSvcVars{
+		name:    o.name,
+		envName: envs[0].Name, // Use any environment to get a sample CloudFormation template for the service.
+		appName: o.appName,
+	})
+	if err != nil {
+		return nil, err
+	}
+	cmd.templateWriter = tplBuf
+	return cmd, nil
+}
+
+func (o *overrideSvcOpts) askIaCTool() error {
```

Comment:
Design/UX-wise, is it better to acknowledge that there is currently only one valid IaCTool? (Like how we surface that we found only one env/workload/etc.) In other words, is it better to do this-- lay the groundwork for eventual multiple choices and ask them to select from one, or is it better to avoid seeming like it could be an oversight/robotic automation?

Reply:
Yea that totally makes sense, we should just default to the one option.
Honestly, I thought that `prompt.MultiSelect` and `prompt.SelectOne` already did that 😂 

What do you think if I update in a separate PR the `prompt` pkg's functions to immediately return the one option if the slice contains one element?